### PR TITLE
Handle thread interruption when enqueueing log message

### LINKED ISSUE #4333: Nextcloud bug at boot
************ CAUSE OF ERROR ************

java.lang.InterruptedException
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireInterruptibly(AbstractQueuedSynchronizer.java:1244)
	at java.util.concurrent.locks.ReentrantLock.lockInterruptibly(ReentrantLock.java:336)
	at java.util.concurrent.LinkedBlockingQueue.offer(LinkedBlockingQueue.java:380)
	at com.nextcloud.client.logger.LoggerImpl.enqueue(LoggerImpl.kt:105)
	at com.nextcloud.client.logger.LoggerImpl.d(LoggerImpl.kt:67)
	at com.nextcloud.client.logger.LegacyLoggerAdapter.i(LegacyLoggerAdapter.kt:31)
	at com.owncloud.android.lib.common.utils.Log_OC.i(Log_OC.java:124)
	at com.owncloud.android.operations.RefreshFolderOperation.checkForChanges(RefreshFolderOperation.java:312)
	at com.owncloud.android.operations.RefreshFolderOperation.run(RefreshFolderOperation.java:220)
	at com.owncloud.android.lib.common.operations.RemoteOperation.execute(RemoteOperation.java:138)
	at com.owncloud.android.syncadapter.FileSyncAdapter.synchronizeFolder(FileSyncAdapter.java:268)
	at com.owncloud.android.syncadapter.FileSyncAdapter.syncChildren(FileSyncAdapter.java:344)
	at com.owncloud.android.syncadapter.FileSyncAdapter.synchronizeFolder(FileSyncAdapter.java:288)
	at com.owncloud.android.syncadapter.FileSyncAdapter.syncChildren(FileSyncAdapter.java:344)
	at com.owncloud.android.syncadapter.FileSyncAdapter.synchronizeFolder(FileSyncAdapter.java:288)
	at com.owncloud.android.syncadapter.FileSyncAdapter.syncChildren(FileSyncAdapter.java:344)
	at com.owncloud.android.syncadapter.FileSyncAdapter.synchronizeFolder(FileSyncAdapter.java:288)
	at com.owncloud.android.syncadapter.FileSyncAdapter.syncChildren(FileSyncAdapter.java:344)
	at com.owncloud.android.syncadapter.FileSyncAdapter.synchronizeFolder(FileSyncAdapter.java:288)
	at com.owncloud.android.syncadapter.FileSyncAdapter.syncChildren(FileSyncAdapter.java:344)
	at com.owncloud.android.syncadapter.FileSyncAdapter.synchronizeFolder(FileSyncAdapter.java:288)
	at com.owncloud.android.syncadapter.FileSyncAdapter.syncChildren(FileSyncAdapter.java:344)
	at com.owncloud.android.syncadapter.FileSyncAdapter.synchronizeFolder(FileSyncAdapter.java:288)
	at com.owncloud.android.syncadapter.FileSyncAdapter.onPerformSync(FileSyncAdapter.java:180)
	at android.content.AbstractThreadedSyncAdapter$SyncThread.run(AbstractThreadedSyncAdapter.java:334)

************ APP INFORMATION ************
ID: com.nextcloud.android.beta
Version: 20190810
Build flavor: versionDev

************ DEVICE INFORMATION ************
Brand: OnePlus
Device: OnePlus5T
Model: ONEPLUS A5010
Id: PQ3A.190605.003
Product: OnePlus5T

************ FIRMWARE ************
SDK: 28
Release: 9
Incremental: 7ec472ea7a

### PR DESCRIPTION
**Pull request by ezaquarii**
Fixes #4333

Signed-off-by: Chris Narkiewicz <hello@ezaquarii.com>

### CODE REVIEW

**Comment by ezaquarii** [comment]
@tobiasKaminsky  I still need to rig a test for this, but feel free to merge it. I'll follow up.

**Comment by AndyScherzinger** [comment]
> @tobiasKaminsky I still need to rig a test for this, but feel free to merge it. I'll follow up.

He's likely offline already and will be back on monday. @tobiasKaminsky I also think this nneeds to be merged _before_ 3.8.0 RC1 (should be part of it)

File: src/main/java/com/nextcloud/client/logger/LoggerImpl.kt [hunk 1]
```diff
@@ -102,11 +102,16 @@ internal class LoggerImpl(
...
-            missedLogs.set(true)
-            missedLogsCount.incrementAndGet()
+        try {
+            val enqueued = eventQueue.offer(entry, 1, TimeUnit.SECONDS)
+            if (!enqueued) {
+                missedLogs.set(true)
+                missedLogsCount.incrementAndGet()
+            }
+        } catch (ex: InterruptedException) {
+            // ignore it as a caller thread can be interrupted we we genuinely want to stop in such case
```
Comment (AndyScherzinger) [inline src/main/java/com/nextcloud/client/logger/LoggerImpl.kt]: too many "we"s ;)

Comment (ezaquarii) [reply src/main/java/com/nextcloud/client/logger/LoggerImpl.kt]: wewe fixed

**Comment by AndyScherzinger** [comment]
cc @tobiasKaminsky for review/merge

**Pull request merged**